### PR TITLE
feat(lists): lists tRPC + @pops/app-lists-db extraction (PRD-140 part API)

### PR DESCRIPTION
**Pull request by knoxio**
## Summary

First slice of PRD-140 (generic lists CRUD UI). Mirrors the 122-API blueprint — extracts the db subtree of `@pops/app-lists` into a new backend-only `@pops/app-lists-db` package so `pops-api` can import services without dragging in the React-bound `@pops/app-lists` ESM source.

Ships the full 14-procedure `listsRouter` under `apps/pops-api/src/modules/lists/router.ts`:

- `lists.list.{list, get, create, update, archive, unarchive, delete}`
- `lists.items.{add, bulkAdd, update, check, uncheck, remove, reorder}`

`lists.list.list` is the router-owned aggregate query (left join `lists` to `list_items`, single round-trip, `GROUP BY l.id`, computes `itemCount` / `uncheckedCount` / `lastUpdatedAt`; supports `kinds[]` + `includeArchived` filters and `updated` / `name` / `created` sort modes). Every other procedure is a thin transactional pass-through to PRD-112 services. `reorder` validates count AND id-set parity against the live item set; `check` / `update` / `uncheck` go through typed `ListNotFoundError` / `ListItemNotFoundError` for `NOT_FOUND` mapping; FK violations on `items.add` surface as `CONFLICT`.

`@pops/app-lists-db` ships as a dist-built node package (`tsc`, Node16) with deps on `@pops/db-types` + `drizzle-orm` only. `@pops/app-lists` keeps the existing `/db` subpath alive via a one-line shim that re-exports the new package, so the in-flight `@pops/app-food-db` seed imports keep resolving while the migration completes. `app-food-db`'s `package.json` now depends on `app-lists-db` directly (the seed step + types pulled the import path through).

Wires PRD-139's backend stub (was `listsRouter = router({})`) up to the real router; updates `router.test.ts`'s `expectedTops` set to include the `lists` id (was intentionally excluded while the router was empty).

## Scope (PRD-140 §AC)

Ticked in this PR:

- [x] All tRPC procedures exist in `apps/pops-api/src/modules/lists/router.ts`.
- [x] All mutations are transactional.
- [x] `lists.list.list` includes computed `itemCount`, `uncheckedCount`, `lastUpdatedAt`.
- [x] `lists.items.reorder` rejects when `orderedIds.length` differs from the live item set OR an id is from a different list.
- [x] PRD-139's manifest backend slot now points at the real router.
- [x] Integration suite at `apps/pops-api/src/modules/lists/__tests__/lists-router.test.ts` covers every procedure.

Deferred to follow-up PRs (140-B index page, 140-C detail page):

- ListsIndexPage + filter chips + create modal
- ListDetailPage + drag reorder + edit modal + delete confirm
- `packages/app-lists/src/pages/__tests__/*.test.tsx` RTL coverage
- E2E flow (create → add items → check → reorder → archive → restore)

## CI / Docker wiring

- `_pkg-check.yml` gains the `app-lists-db` build step in both the typecheck and test phases (alongside the `food-contracts` step PRD-125 added).
- `pops-api` / `pops-shell` / `pops-mcp` Dockerfiles copy + build the new package alongside `app-food-db` and `food-contracts`.

## Local results

- `@pops/api` integration: 35/35 (lists-router 29 + router.test 6) on the rebased branch; full `@pops/api` suite 4545/4545 pre-rebase.
- `@pops/app-lists-db`: 35/35.
- `@pops/app-lists`: 9/9.
- `@pops/app-food-db`: 203/203.
- Typecheck clean on every package I touched. The 9 pre-existing `PillarRegistryEntry` / `pillar-unavailable` errors in `pops-api` come from pre-flight P2 (#2715) on main; they're not in `lists/*` files and exist on a clean checkout of main.

## Test plan

- [ ] `pnpm install` then `pnpm --filter @pops/api test` runs the lists router suite end-to-end against an in-memory SQLite seeded with `0062_chemical_donald_blake.sql`.
- [ ] `pnpm --filter @pops/app-lists-db test` covers the PRD-112 service-layer invariants (now in the new package).
- [ ] `pnpm --filter @pops/app-food-db test` continues to pass (seed imports updated to `@pops/app-lists-db`).
- [ ] CI's `_pkg-check.yml` builds `@pops/app-lists-db` before per-package typecheck/test runs.
- [ ] Docker builds for `pops-api` / `pops-shell` / `pops-mcp` pick up the new package via the Dockerfile updates.

## Coordination

Strictly additive. Only shared files touched: `apps/pops-api/src/router.test.ts` (one-line `expectedTops` extension), `apps/pops-api/src/modules/lists/index.ts` (manifest wiring), `_pkg-check.yml` + the three Dockerfiles (CI build wiring). No food-domain files touched. Sibling to every in-flight food PR.

### CODE REVIEW

File: apps/pops-api/src/modules/lists/routers/list.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+import {
+  archiveList,
+  createList,
+  deleteList,
+  getList,
+  listItemsForList,
+  unarchiveList,
+  updateList,
+} from '@pops/app-lists-db';
```
[the remark this answers is not in the document]
Comment (knoxio) [reply apps/pops-api/src/modules/lists/routers/list.ts:21]: Addressed in b3bd70b3 — imported `ListNotFoundError` from `@pops/app-lists-db` and switched the catch branch to `instanceof`.

File: apps/pops-api/src/modules/lists/routers/list.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    try {
+      updateList(getDrizzle(), input.id, { name: input.name, kind: input.kind });
+      return { ok: true as const };
+    } catch (err) {
+      if (err instanceof Error && err.name === 'ListNotFoundError') {
+        return { ok: false as const, reason: 'NotFound' as const };
+      }
+      throw err;
+    }
+  }),
```
[the remark this answers is not in the document]
Comment (knoxio) [reply apps/pops-api/src/modules/lists/routers/list.ts:107]: Same fix as #3381548387 — catch now uses `err instanceof ListNotFoundError`.

File: apps/pops-api/src/modules/lists/routers/items.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    if (current.length !== input.orderedIds.length) {
+      return { ok: false as const, reason: 'BadIds' as const };
+    }
+    const currentSet = new Set(current);
+    for (const id of input.orderedIds) {
+      if (!currentSet.has(id)) return { ok: false as const, reason: 'BadIds' as const };
+    }
+    runOrMap(() => reorderItems(db, input.listId, input.orderedIds));
+    return { ok: true as const };
+  }),
```
[the remark this answers is not in the document]
Comment (knoxio) [reply apps/pops-api/src/modules/lists/routers/items.ts:141]: Addressed in b3bd70b3 — added a Set-based duplicate-id check before the membership scan. `orderedIds` now has to be a true permutation of the live item set (same length, every id belongs to the list, no duplicates). New router test covers `[a, a]` rejection and asserts positions are untouched.

File: apps/pops-api/src/modules/lists/services/aggregate.ts [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+interface RawRow {
+  id: number;
+  name: string;
+  kind: ListKind;
+  owner_app: string;
+  archived_at: string | null;
+  item_count: number;
+  unchecked_count: number;
+  last_updated_at: string;
+}
```
[the remark this answers is not in the document]
Comment (knoxio) [reply apps/pops-api/src/modules/lists/services/aggregate.ts:58]: Addressed in b3bd70b3 — `unchecked_count` is now typed `number | null` in `RawRow` with an inline comment noting the SUM-over-empty-LEFT-JOIN NULL. Mapping still normalises to 0.

**Comment by knoxio** [comment]
Re: the `archiveList` semantics issue called out in the review overview — addressed in b3bd70b3. The service now always sets `archived_at = nowIso()` on archive, including when the row is already archived (matches PRD-140 §Edge Cases "archive again updates archived_at to now"). The PRD-112 service test in `@pops/app-lists-db` was updated to assert the new "refresh timestamp" semantics with a sub-ms spin so the second timestamp is provably newer than the first.

Also fixed CI: the per-app workflows (`api-quality.yml`, `api-test.yml`, `fe-quality.yml`, `fe-test-e2e.yml`) have their own hardcoded build sequence (separate from `_pkg-check.yml`) which I missed in the first pass. `@pops/app-lists-db` now sits between `food-contracts` and `app-food-db` in all four. That's why the first run hit `Cannot find module '@pops/app-lists-db'` during the `app-food-db` build — the dep wasn't built yet.
